scripts/docs-generation/3rdPartyDocs.js

```diff
@@ -0,0 +1,109 @@
+const fs = require('fs')
+const path = require('path')
+const request = require('request')
+
+const { buildPreface } = require('../utils/helpers')
+const reporters3rdParty = require('./3rd-party/reporters.json')
+const services3rdParty = require('./3rd-party/services.json')
+
+const packages3rdParty = [...reporters3rdParty, ...services3rdParty]
+const plugins = {
+    reporter: ['Reporter', 'Reporter'],
+    service: ['Services', 'Service']
+}
+const githubHost = 'https://github.com/'
+const githubRawHost = 'https://raw.githubusercontent.com/'
+const githubReadme = 'master/README.md'
+
+const readmeHeaderLines = 9
+const readmeHeaders = ['===', '# ']
+const readmeBadges = 'https://badge'
+
+/**
+ * Generate docs for 3rd party reporters and services
+ * @param {object} sidebars website/sidebars
+ */
+exports.generate3rdPartyDocs = async (sidebars) => {
+    for (const [, [namePlural, nameSingular]] of Object.entries(plugins)) {
+        for (const { packageName, title, githubUrl } of packages3rdParty) {
+            const readme = await downloadReadme(githubUrl)
+            const id = `3rd-party-${packageName}`.replace(/@/g, '').replace(/\//g, '-')
+
+            const doc = normalizeDoc(readme, githubUrl, buildPreface(id, title, nameSingular, `${githubUrl}/edit/${githubReadme}`))
+            fs.writeFileSync(path.join(__dirname, '..', '..', 'docs', `_${id}.md`), doc, { encoding: 'utf-8' })
+
+            if (!sidebars.docs[namePlural]) {
+                sidebars.docs[namePlural] = []
+            }
+
+            // eslint-disable-next-line no-console
+            console.log(`Generated docs for ${packageName}`)
+
+            sidebars.docs[namePlural].push(id)
```

Comment:
Currently reporters and services are pushed into both buckets:

```json
"Reporter": [
      "allure-reporter",
      "concise-reporter",
      "dot-reporter",
      "junit-reporter",
      "spec-reporter",
      "sumologic-reporter",
      "3rd-party-wdio-reportportal-reporter",
      "3rd-party-wdio-video-reporter",
      "3rd-party-rpii-wdio-html-reporter",
      "3rd-party-wdio-json-reporter",
      "3rd-party-wdio-mochawesome-reporter",
      "3rd-party-wdio-timeline-reporter",
      "3rd-party-wdio-chromedriver-service",
      "3rd-party-wdio-intercept-service",
      "3rd-party-wdio-zafira-listener-service",
      "3rd-party-wdio-reportportal-service",
      "3rd-party-wdio-docker-service"
    ],
    "Services": [
      "appium-service",
      "applitools-service",
      "browserstack-service",
      "devtools-service",
      "firefox-profile-service",
      "sauce-service",
      "selenium-standalone-service",
      "testingbot-service",
      "3rd-party-wdio-reportportal-reporter",
      "3rd-party-wdio-video-reporter",
      "3rd-party-rpii-wdio-html-reporter",
      "3rd-party-wdio-json-reporter",
      "3rd-party-wdio-mochawesome-reporter",
      "3rd-party-wdio-timeline-reporter",
      "3rd-party-wdio-chromedriver-service",
      "3rd-party-wdio-intercept-service",
      "3rd-party-wdio-zafira-listener-service",
      "3rd-party-wdio-reportportal-service",
      "3rd-party-wdio-docker-service"
    ]
```
which shows everything in the service section:
<img width="222" alt="Screen Shot 2019-05-27 at 9 55 34 AM" src="https://user-images.githubusercontent.com/731337/58401097-9ae10080-8065-11e9-8bb7-cc9fe51ec9c2.png">